internal/services/recoveryservices/recovery_services_vault_resource.go

```diff
@@ -32,7 +32,7 @@ func resourceRecoveryServicesVault() *pluginsdk.Resource {
 		Delete: resourceRecoveryServicesVaultDelete,
 
 		Importer: pluginsdk.ImporterValidatingResourceId(func(id string) error {
-			_, err := parse.VaultID(id)
```

Comment:
can we remove the existing ID parser? do we need a state migration from one to the other?

Reply:
checked the logic in original parse and new one, it seems to be the same (especially in casing) so I think we don't need a migration